www/js/lib/cookies.js

```diff
@@ -21,11 +21,43 @@ define([], function() {
 |*|
 \*/
 
+var storeType = testStorageSupport();
+
+// Tests for localStorage support
+function testStorageSupport() {
+  // Prefer localStorage if supported due to some platforms blocking cookies in local contexts
+  // DEV: In FF extensions, cookies are blocked since at least FF 68.6 but possibly since FF 55 [kiwix-js #612]
+  var type = 'local_storage';
+  // First test for localStorage support
+  var localStorageTest = false;
+  try {
+    localStorageTest = 'localStorage' in window && window['localStorage'] !== null;
+    // DEV: Above test returns true in IE11 running from file:// protocol, but attempting to write a key to
+    // localStorage causes an exception; so to test fully, we must now attempt to write and remove a test key
+    if (localStorageTest) {
+      localStorage.setItem('kiwixStorage', '');
+      localStorage.removeItem('kiwixStorage');
+    }
+  } catch (e) {
+    localStorageTest = false;
+    console.log('LocalStorage is not supported!');
+  }
+  // Fall back to cookie if localStorage fails
+  if (!localStorageTest) type = 'cookie';
+  console.log('Storage test: type: ' + type);
```

Comment:
Maybe it would be better to report this status in the Settings section of the app (just like the other APIs status).
I realize it's more complicated : it might be a separate enhancement github issue if you prefer.

Reply:
I can do this.

Reply:
Done.